rudder-core/src/main/scala/com/normation/rudder/services/system/DatabaseManager.scala

```diff
@@ -124,24 +124,44 @@ class DatabaseManagerImpl(
      reportsRepository.archiveEntries(date)
    }
 
+
    def deleteEntries(date : DateTime) : Box[Int] = {
      val reports = reportsRepository.deleteEntries(date)
      val nodeConfigs = expectedReportsRepo.deleteNodeConfigIdInfo(date)
-
-     (reports, nodeConfigs) match {
-       case (Full(reports), Full(nodeConfig)) => Full(reports+nodeConfig)
-       case (b:EmptyBox,Full(_)) => b
-       case (Full(_), b:EmptyBox) => b
-       case (a:EmptyBox, b:EmptyBox) =>
-         val reportError = a match {
-           case f:Failure => f
-           case Empty => Failure("An error occured while deleting reports", Empty, Empty)
-         }
-         val nodeConfigError = b match {
-           case f:Failure => f
-           case Empty => Failure("An error occured while deleting old nodeConfig", Empty, Empty)
-         }
-         Failure(nodeConfigError.msg, nodeConfigError.exception, Full(reportError))
+     val expectedReports = expectedReportsRepo.deleteExpectedReports(date)
+
+     // Maybe we could use a reduce here
+     (reports, nodeConfigs, expectedReports) match {
```

Comment:
I allow myself to do that because I prevent Empty case:


     val reports = reportsRepository.deleteEntries(date) ?~! "An error occured while deleting reports"
     val nodeConfigs = expectedReportsRepo.deleteNodeConfigIdInfo(date) ?~! "An error occured while deleting reports"
     val expectedReports = expectedReportsRepo.deleteExpectedReports(date) ?~! "An error occured while old expected reports"